### PR TITLE
Add ReadbackSpan which makes stagemonitor independent of the OT impl

### PR DESCRIPTION
**Pull request by felixbarny**
/cc @adriancole

### CODE REVIEW

File: stagemonitor-requestmonitor/src/test/java/org/stagemonitor/requestmonitor/tracing/B3Propagator.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+
+public class B3Propagator implements MockTracer.Propagator {
+	@Override
+	public <C> void inject(MockSpan.MockContext ctx, Format<C> format, C carrier) {
+		if (format instanceof B3HeaderFormat) {
+			((TextMap) carrier).put(B3HeaderFormat.SPAN_ID_NAME, Long.toHexString(ctx.spanId()));
```
Comment (adriancole) [inline stagemonitor-requestmonitor/src/test/java/org/stagemonitor/requestmonitor/tracing/B3Propagator.java]: probably also want parent and sampled, right?

**Comment by adriancole** [comment]
for brave, if you want to create a new type of format token, I'd advise against using mock classes.. maybe this instead?
* we could expose BraveTracer.INJECTOR public as BraveTracer.B3_INJECTOR
* add a builder with `Builder add(Format<TextMap> format, TraceContext.Injector<TextMap> injector)`;
* then you can do `BraveTracer.newBuilder(tracer).add(CUSTOM_TEXTMAP, BraveTracer.B3_INJECTOR).build();`

cc @devinsba wdyt?

File: stagemonitor-requestmonitor/src/test/java/org/stagemonitor/requestmonitor/tracing/B3Propagator.java (hunk 1 above)
Comment (felixbarny) [inline stagemonitor-requestmonitor/src/test/java/org/stagemonitor/requestmonitor/tracing/B3Propagator.java]: Don't know how to access them. However it is good enough for unit tests. This is only for the MockTracer afer all

Comment (adriancole) [inline stagemonitor-requestmonitor/src/test/java/org/stagemonitor/requestmonitor/tracing/B3Propagator.java]: ahh sorry didn't see test in that package name!

**Pull request merged**